src/Core/Scheduler/Providers/WMS_Provider.js

```diff
@@ -7,6 +7,78 @@
 import * as THREE from 'three';
 import Extent from '../../Geographic/Extent';
 import OGCWebServiceHelper from './OGCWebServiceHelper';
+import Fetcher from './Fetcher';
+
+const mimeTypeByTagName = {
+    JPEG: 'image/jpeg',
+    PNG: 'image/png',
+    GIF: 'image/gif',
+    PPM: 'image/x‑portable‑pixmap',
+    TIFF: 'image/tiff',
+    GeoTIFF: 'image/tiff',
+    WebCGM: 'image/cgm;Version=4;ProfileId=WebCGM',
+    SVG: 'image/svg+xml',
+    WMS_XML: 'application/vnd.ogc.wms_xml',
+    'GML.1': 'application/gml+xml',
+    'GML.2': 'application/gml+xml',
+    'GML.3': 'application/gml+xml',
+    WBMP: 'image/vnd.wap.wbmp',
+    MIME: 'www/mime', // Not sure of this one (there's also message/rfc822)
+};
+/* For reference, there are also these 2 possible tags that do not translate to mimetype
+   INIMAGE display text in the returned image -->
+   BLANK return an image with all pixels transparent if
+   */
+
+function _parseSupportedCrs(version, xmlLayer) {
+    let supportedCrs = [];
+    const crsPropName = version === '1.3.0' ? 'CRS' : 'SRS';
+
+    for (const childElem of xmlLayer.children) {
+        if (childElem.tagName === crsPropName) {
+            supportedCrs = supportedCrs.concat(childElem.textContent.trim().split(' '));
+        }
+    }
+    if (xmlLayer.parentNode && xmlLayer.parentNode.tagName === 'Layer') {
+        supportedCrs = supportedCrs.concat(_parseSupportedCrs(version, xmlLayer.parentNode));
+    }
+
+    return supportedCrs;
+}
+
+function parseLayerBoundingBox(version, targetCrs, xmlLayer) {
+    const crsPropName = version === '1.3.0' ? 'CRS' : 'SRS';
+    for (const childElem of xmlLayer.children) {
+        if (childElem.tagName === 'BoundingBox' && (!targetCrs || targetCrs === childElem.getAttribute(crsPropName))) {
+            return new Extent(
+                    childElem.getAttribute(crsPropName),
+                    childElem.getAttribute('minx'),
+                    childElem.getAttribute('maxx'),
+                    childElem.getAttribute('miny'),
+                    childElem.getAttribute('maxy'));
+        }
+    }
+    if (xmlLayer.parentNode && xmlLayer.parentNode.tagName === 'Layer') {
+        return parseLayerBoundingBox(version, targetCrs, xmlLayer.parentNode);
+    }
+}
+
+function parseGeoBounds(version, xmlLayer) {
+    const geoTagName = version === '1.3.0' ? 'EX_GeographicBoundingBox' : 'LatLonBoundingBox';
```

Comment:
you could group in private functions 
```js
getCrsPNByVersion(version)...

getGeoPNByVersion(version)...
```

Reply:
You want a function for a one-liner?

Reply:
it's just to group the constants
you can suggest something else

Reply:
Oh I see. Yes, a function should be better than constants here I guess.